error.go

```diff
@@ -491,17 +493,19 @@ func (e Error) Description() string {
 }
 
 func isTimeout(err error) bool {
-	e, ok := err.(interface {
-		Timeout() bool
-	})
-	return ok && e.Timeout()
+	var opError net.Error
```

Comment:
the other option that is a _bit_ closer to your previous implementation is to unwrap into an interface that only has timeout

```
func isTimeout(err error) bool {
	var timeoutError interface{ Timeout() bool }
	if errors.As(err, &timeoutError) {
		return timeoutError.Timeout()
	}
	return false
}

func isTemporary(err error) bool {
	var tempError interface{ Temporary() bool }
	if errors.As(err, &tempError) {
		return tempError.Temporary()
	}
	return false
}
```

Reply:
This is what I would prefer, we can add handling for other error types (eg: `net.Error`) as well, but let's not fundamentally change the logic here to start with.

Reply:
@dominicbarnes updated, lemme know what you think